src/edu/csus/ecs/pc2/imports/ccs/ICPCTSVLoader.java

```diff
@@ -319,17 +329,57 @@ public static Group[] loadGroups(String filename) throws Exception {
             } catch (NumberFormatException e) {
                 throw new InvalidValueException("Expecting group number got '" + numberStr + "' on line " + lines[i], e);
             }
-
-            Group group = new Group(groupName);
+            
+            Group group = findGroupById(authoritativeGroups, number);
+            if (group == null) {
+                group = new Group(groupName);
+            }
+            
+            group.setDisplayName(groupName);
             group.setGroupId(number);
-
-            groups[groupAccount] = group;
-            groupAccount++;
+            
+            groupList.add(group);
         }
-
+        
+        /**
+         * Add groups from authoritativeGroups if missing from groupList.
+         */
+        for (Group aGroup : authoritativeGroups) {
+            
+            Group missingGroup = null; 
+            
+            for (Group glGroup : groupList) {
+                if (aGroup.getGroupId() == glGroup.getGroupId()) {
+                    missingGroup = glGroup;
+                }
+            }
+            
+            if (missingGroup == null) {
+                groupList.add(aGroup);
+            }
+        }
+        
+        groups = (Group[]) groupList.toArray(new Group[groupList.size()]);
+        
         return groups;
     }
-    
+
+    /**
+     * Search and return group if found.
+     * 
+     * @param authoritativeGroups list of groups
+     * @param groupCMSId group (CMS
+     * @return null if not found, else the group
```

Comment:
What happened to the rest of the comment here?

Reply:
java doc updated.